scripts/loadEmptyKeys.js

```diff
@@ -0,0 +1,137 @@
+//
+// See this JIRA for more information: https://issues.gpii.net/browse/GPII-3381
+//
+"use strict";
+
+var process = require("process"),
+    fs = require("fs"),
+    fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    uuid = uuid || require("node-uuid");
+
+require("./shared/prefsSetsDbUtils.js");
+require("gpii-pouchdb");
+
+fluid.defaults("gpii.uuidLoader", {
+    gradeNames: ["gpii.pouch"],
+    outputFileName: process.env.OUTPUT_FILENAME || "generated-keys-" + new Date().toISOString() + ".txt",
+    totalNumOfKeys: parseInt(process.env.NUM_OF_KEYS),
+    hideProgress: process.env.HIDE_PROGRESS,
+    count: 1,
+    dbOptions: {
+        name: process.env.COUCHDB_URL
+    },
+    invokers: {
+        addKeyInDb: {
+            funcName: "gpii.uuidLoader.addKeyInDb",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+        },
+        createNewKeys: {
+            funcName: "gpii.uuidLoader.createNewKeys",
+            args: [
+                "{that}",
+                "{that}.options.totalNumOfKeys",
+                "{that}.options.outputFileName",
+                "{that}.options.hideProgress",
+                "{that}.options.count"
+            ]
+        },
+        dbPut: {
+            funcName: "gpii.uuidLoader.dbPut",
+            args: ["{that}", "{arguments}.0"]
+        }
+    }
+});
+
+gpii.uuidLoader.dbPut = function (that, doc) {
+    var promise = fluid.promise();
+    // We only perform the put after checking that the uuid doesn't exist in DB,
+    // just in case somebody wants to reuse this in the future.
+    //
+    that.get(doc._id).then(function (/* record */) {
+        promise.reject({isError: true, message: "doc with id " + doc._id + " already exists in DB"});
+    }, function (err) {
+        if (err.message === "missing") {
+            that.put(doc).then(function (newDoc) {
+                promise.resolve(newDoc);
+            }, function (err) {
+                promise.reject(err);
+            });
+        } else {
+            promise.reject(err);
+        };
+    });
+    return promise;
+};
+
+gpii.uuidLoader.addKeyInDb = function (that, keyData) {
+    var promise = fluid.promise();
+    var sequence = [
+        that.dbPut(keyData.gpiiKey),
+        that.dbPut(keyData.prefsSafe)
+    ];
+
+    fluid.promise.sequence(sequence).then(function () {
+        promise.resolve(keyData.gpiiKey._id);
+    }, function (error) {
+        promise.reject(error);
+    });
+
+    return promise;
+};
+
+// The real action
+//
+gpii.uuidLoader.createNewKeys = function (that, totalNumOfKeys, outputFileName, hideProgress, count) {
+    var keyData = gpii.prefsSetsDbUtils.generateKeyData(uuid.v4());
+    that.addKeyInDb(keyData).then(function (newKey) {
+        fs.appendFileSync(outputFileName, newKey + "\n");
+        if (!hideProgress) {
+            console.log("Key", count, "of", totalNumOfKeys, "created:", newKey);
```

Comment:
Add spaces in between words for readability.

Reply:
@cindyli,

when we run the script you get the output like this: "Key 2 of 20 created: bdd229c3-e41e-4129-b632-de22bfd381e5"

Is this wrong? ... OR do you mean in code by appending the strings with "+" rather than with ","?

Reply:
This message looks great. Nothing needs to be changed. Sorry about the comment.

Reply:
Okay, no prob, just wanted to double check.